compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/pager/Pager.kt

```diff
@@ -956,7 +956,11 @@ private inline fun debugLog(generateMsg: () -> String) {
 }
 
 @OptIn(ExperimentalFoundationApi::class)
-private fun PagerState.isScrollingForward() = dragGestureDelta() < 0
+private fun PagerState.isScrollingForward(velocity: Float) = if (isNotGestureAction()) {
+    velocity
```

Comment:
Shouldn't we just use `velocity` in all cases?

Reply:
This component contains a few checks if the direction of scroll doesn't match the direction of drag, so it seems much safer to add a fallback if `dragGestureDelta` is zero instead of completely replacing the current logic.